### PR TITLE
 feat: [M3-6749] - Subnet delete dialog

### PR DESCRIPTION
**Pull request by coliu-akamai**
## Description 📝
- add subnet delete dialog to enable deleting of subnets

## Preview 📷

Delete disabled on action menu:
image

Subnet Delete Dialog:
image

(tooltip icon color is off in this video recording again -- the color kinda just changes randomly 😭 )
https://github.com/linode/manager/assets/139280159/471dc77b-1975-4b08-bdaa-621df7023022

## How to test 🧪
```
yarn test SubnetDeleteDialog.test.tsx
yarn test VPCSubnetsTable.test.tsx
yarn cy:run -s "cypress/e2e/core/vpc/vpc-details-page.spec.ts"
```
Navigate to a VPC's detail page and:
- Verify that you cannot delete subnets that have linodes assigned to them
- Verify that you can delete subnets that have no linodes assigned to them

### CODE REVIEW

File: packages/manager/src/features/VPCs/VPCDetail/SubnetActionMenu.tsx [hunk 1]
```diff
@@ -3,14 +3,30 @@ import * as React from 'react';
...
 
   const handleUnassignLinode = () => {};
 
   const handleEdit = () => {};
 
-  const handleDelete = () => {};
+  const handleDelete = () => {
+    const { handleDelete } = props;
+    handleDelete(subnetId, subnetLabel);
+  };
```
Comment (hana-linode) [inline packages/manager/src/features/VPCs/VPCDetail/SubnetActionMenu.tsx:29]: We can remove this unnecessary abstraction and destrcuture `handleDelete` along with the other props on line 18, then update the actions onClick to `handleDelete(subnetId, subnetLabel);`

File: packages/manager/src/features/VPCs/VPCDetail/SubnetDeleteDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  const onDeleteSubnet = async () => {
+    await deleteSubnet();
+    enqueueSnackbar('Subnet deleted successfully.', {
+      variant: 'success',
+    });
+    onClose();
+    if (history.location.pathname !== `/vpcs/${vpcId}`) {
+      history.push(`/vpcs/${vpcId}`);
+    }
```
[the remark this answers is not in the document]
Comment (coliu-akamai) [reply packages/manager/src/features/VPCs/VPCDetail/SubnetDeleteDialog.tsx:34]: looking through the vpc UI designs, don't think so - will remove, thanks for catching!

File: packages/manager/src/features/VPCs/VPCDetail/VPCDetail.tsx [hunk 3]
```diff
@@ -184,7 +199,19 @@ const VPCDetail = () => {
...
+          setDeleteSubnetDialogData((subnetDialogData) => ({
+            ...subnetDialogData,
+            open: false,
+          }))
+        }
+        open={deleteSubnetDialogData.open}
+        subnetId={deleteSubnetDialogData.subnetId}
+        subnetLabel={deleteSubnetDialogData.subnetLabel}
+        vpcId={vpc.id}
+      />
```
Comment (hana-linode) [inline packages/manager/src/features/VPCs/VPCDetail/VPCDetail.tsx:214]: Can we move the subnets logic to the `VPCSubnetsTable` component?

Comment (coliu-akamai) [reply packages/manager/src/features/VPCs/VPCDetail/VPCDetail.tsx]: done + moved related tests to new SubnetActionMenu test file

File: packages/manager/src/features/VPCs/VPCDetail/VPCSubnetsTable.tsx [hunk 4]
```diff
@@ -30,9 +31,16 @@ interface Props {
...
-export const VPCSubnetsTable = ({ vpcId }: Props) => {
+export const VPCSubnetsTable = (props: Props) => {
+  const { vpcId } = props;
   const [subnetsFilterText, setSubnetsFilterText] = React.useState('');
 
+  const [deleteSubnetDialogData, setDeleteSubnetDialogData] = React.useState<{
+    open: boolean;
+    subnetId?: number;
+    subnetLabel?: string;
+  }>({ open: false, subnetId: -1 });
```
Comment (hana-linode) [inline packages/manager/src/features/VPCs/VPCDetail/VPCSubnetsTable.tsx:42]: We can simplify this state by separating it into two different states: dialog open state and selected subnet state (see `VPCLanding` for example)

Comment (coliu-akamai) [reply packages/manager/src/features/VPCs/VPCDetail/VPCSubnetsTable.tsx]: that's definitely a lot cleaner, thanks! added :D

**Review by hana-linode — APPROVED** [review]
Nice work! 🚀

**Review by jdamore-linode — APPROVED** [review]
Nice work, @coliu-akamai!

**Comment by coliu-akamai** [comment]
Merging as integration test failures are unrelated to this PR!

**Pull request merged**